common/src/main/java/org/vivecraft/common/CustomShapedRecipe.java

```diff
@@ -29,18 +30,18 @@ public class CustomShapedRecipe {
             ExtraCodecs
                 .strictOptionalField(ExtraCodecs.POSITIVE_INT, "hideflags", 0)
                 .forGetter(itemStack -> itemStack.getOrCreateTag().getInt("HideFlags"))
-        ).apply(instance, (item, vanillaitem, count, name, unbreakable, hideflags) -> {
-            if (vanillaitem != Items.AIR) {
-                ItemStack itemStack = new ItemStack(vanillaitem, count);
-                if (!name.getString().isEmpty()) {
-                    itemStack.setHoverName(Component.translatable(name.getString()));
-                }
-                itemStack.getOrCreateTag().putInt("HideFlags", hideflags);
-                itemStack.getOrCreateTag().putBoolean("Unbreakable", unbreakable);
-                return itemStack;
-            } else {
-                return new ItemStack(item, count);
+        ).apply(instance, (vanillaitem, count, name, unbreakable, hideflags) -> {
+            ItemStack itemStack = new ItemStack(vanillaitem, count);
+            if (!name.getString().isEmpty()) {
+                itemStack.setHoverName(Component.translatable(name.getString()));
             }
+            itemStack.getOrCreateTag().putInt("HideFlags", hideflags);
+            itemStack.getOrCreateTag().putBoolean("Unbreakable", unbreakable);
+            return itemStack;
         })
     );
+
+    public static final Codec<ItemStack> CODEC = ExtraCodecs.either(VIVECRAFT_ITEMSTACK_OBJECT_CODEC, CraftingRecipeCodecs.ITEMSTACK_OBJECT_CODEC)
+        .xmap(itemStackItemStackEither -> itemStackItemStackEither.map(stack -> stack, stack -> stack), Either::right);
```

Comment:
not that familiar with codecs, what does this do?

Reply:
So from what I understand about it, it first tries to use `VIVECRAFT_ITEMSTACK_OBJECT_CODEC` to get an itemstack. But this fails due to missing fields for most items. Then instead it will use `CraftingRecipeCodecs.ITEMSTACK_OBJECT_CODEC` to get the itemstack. Though I will admit I'm not that great with codecs, this does seem to work when testing. Would like it if you also could give it some tests to ensure it works?

Reply:
will do